charts/spv-wallet-stack/charts/spv-wallet/templates/deployment.yml

```
@@ -65,10 +65,15 @@ spec:
                   name: {{ .Release.Name }}-postgresql-spv-wallet
                   key: password
           {{- end }}
-            - name: SPVWALLET_PAYMAIL_BEEF_BLOCK_HEADER_SERVICE_AUTH_TOKEN
+            - name: SPVWALLET_PAYMAIL_BEEF_BLOCK_HEADERS_SERVICE_AUTH_TOKEN
               valueFrom:
                 secretKeyRef:
-                  name: block-headers-service-secret
+                  name: 
+                  {{- if .Values.config.paymail.beef.block_headers_service_auth_token }}
+                    {{ include "spv-wallet.fullname" . }}
+                  {{- else }}
+                    block-headers-service-secret
+                  {{- end }}
```

Minor, this way the value of name is in the same line

```suggestion 
                  {{- if .Values.config.paymail.beef.block_headers_service_auth_token }}
                  name:  {{ include "spv-wallet.fullname" . }}
                  {{- else }}
                  name:  block-headers-service-secret
                  {{- end }}
```